### PR TITLE
Table adapter updates

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/table/src/ToggleExpandedIcon/ToggleExpandedIcon.types.ts [hunk 1]
```diff
@@ -3,6 +3,10 @@ import { BaseIconButtonProps } from '@leafygreen-ui/icon-button';
 import { RowProps } from '../Row/Row.types';
 import { LGRowData } from '../useLeafyGreenTable';
 
+/**
+ * @privateRemarks
```
Comment (TheSonOfThomp) [inline packages/table/src/ToggleExpandedIcon/ToggleExpandedIcon.types.ts:7]: @ internal or deprecated?

**Review by TheSonOfThomp — APPROVED** [review]
👍 Filed this ticket https://jira.mongodb.org/browse/LG-3062

File: packages/table/src/Cell/HeaderCell/HeaderCell.tsx [hunk 2]
```diff
@@ -67,7 +67,7 @@ const HeaderCell = <T extends LGRowData>({
         className={cx(
           cellContentContainerStyles,
           headerCellContentStyles,
-          // @ts-expect-error Header types need to be extended or declared in the react-table namespace
+          // @ts-ignore Header types need to be extended or declared in the react-table namespace
```
Comment (TheSonOfThomp) [inline packages/table/src/Cell/HeaderCell/HeaderCell.tsx:70]: This should still be expect-error, so we can remove it if/when it's fixed

File: packages/table/src/Row/Row.stories.tsx [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
 import React from 'react';
 import { ComponentStory, Meta } from '@storybook/react';
+import {
+  ColumnDef,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  HeaderGroup,
+} from '@tanstack/react-table';
```
Comment (TheSonOfThomp) [inline packages/table/src/Row/Row.stories.tsx:11]: 👍 Definitely a lot clearer seeing what is ours & what is theirs

**Review by TheSonOfThomp — APPROVED** [review]

File: packages/table/src/Cell/HeaderCell/HeaderCell.tsx (hunk 2 above)
Comment (spark33) [reply packages/table/src/Cell/HeaderCell/HeaderCell.tsx:70]: This is one of those errors that doesn't show up locally, but shows up in build. So if we were to keep @ts-expect-error, we'd get an error for an unused directive.

Comment (spark33) [reply packages/table/src/Cell/HeaderCell/HeaderCell.tsx:71]: Updated comment accordingly!

**Pull request merged**